src/main/java/com/novi/springboottechiteasy/services/TelevisionService.java

```diff
@@ -0,0 +1,349 @@
+                    }
+                })
+                .orElseThrow(() -> new RecordNotFoundException("No television found with id: " + id));
+    }
+
+    public TelevisionDto addTelevision(TelevisionInputDto inputDto) {
+        Television television = transferToTelevision(inputDto);
+        televisionRepository.save(television);
+        return televisionDtoMapper.mapToDto(television);
+    }
+
+    public TelevisionDto updateTelevision(Long id, TelevisionInputDto newTelevision) {
+        Optional<Television> television = televisionRepository.findById(id);
+
+        if (television.isPresent()) {
+
+            Television thisTelevision = television.get();
+            thisTelevision.setType(newTelevision.getType());
+            thisTelevision.setBrand(newTelevision.getBrand());
+            thisTelevision.setName(newTelevision.getName());
+            thisTelevision.setPrice(newTelevision.getPrice());
+            thisTelevision.setAvailableSizes(newTelevision.getAvailableSizes());
+            thisTelevision.setRefreshRate(newTelevision.getRefreshRate());
+            thisTelevision.setScreenType(newTelevision.getScreenType());
+            thisTelevision.setScreenQuality(newTelevision.getScreenQuality());
+            thisTelevision.setSmartTv(newTelevision.getSmartTv());
+            thisTelevision.setWifi(newTelevision.getWifi());
+            thisTelevision.setVoiceControl(newTelevision.getVoiceControl());
+            thisTelevision.setHdr(newTelevision.getHdr());
+            thisTelevision.setBluetooth(newTelevision.getBluetooth());
+            thisTelevision.setAmbiLight(newTelevision.getAmbiLight());
+            thisTelevision.setOriginalStock(newTelevision.getOriginalStock());
+            thisTelevision.setOriginalStockDate(newTelevision.getOriginalStockDate());
+            thisTelevision.setSold(newTelevision.getSold());
+            if (newTelevision.getSoldDates() != null) {
+                thisTelevision.getSoldDates().addAll(newTelevision.getSoldDates());
+            }
+
+            Television saveTelevision = televisionRepository.save(thisTelevision);
+
+            return televisionDtoMapper.mapToDto(saveTelevision);
+
+        } else {
+            throw new RecordNotFoundException("No television found with id: " + id);
+        }
+    }
+
+    public void assignRemoteControllerToTelevision(Long id, Long remoteControllerId) {
+        Optional<Television> television = televisionRepository.findById(id);
+        Optional<RemoteController> remote = remoteControllerRepository.findById(remoteControllerId);
+
+        if (television.isPresent() && remote.isPresent()) {
+            Television thisTelevision = television.get();
+            RemoteController thisRemote = remote.get();
+            thisTelevision.setRemoteController(thisRemote);
+            thisRemote.setTelevision(thisTelevision);
+
+            televisionRepository.save(thisTelevision);
+            remoteControllerRepository.save(thisRemote);
```

Comment:
Je hoeft niet beide kanten van de relatie te saven, alleen de owner kant,. In geval van een OneToOne is dat de kant waar het in de tabel staat.